sound/soc/sof/topology.c

```diff
@@ -2876,56 +2884,73 @@ static int sof_link_ssp_load(struct snd_soc_component *scomp, int index,
 {
 	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(scomp);
 	struct snd_soc_tplg_private *private = &cfg->priv;
+	int number_configs = le32_to_cpu(cfg->num_hw_configs);
 	u32 size = sizeof(*config);
+	int current_config;
 	int ret;
+	int i;
 
-	/* handle master/slave and inverted clocks */
-	sof_dai_set_format(hw_config, config);
+	/* get the default config index */
+	for (current_config = 0; current_config < number_configs; current_config++) {
+		if (cfg->hw_config[current_config].id == cfg->default_hw_config_id)
+			break;
+	}
```

Comment:
This loop will always find a match because the caller has already done that once... But then you don't actually need to run this search again, just pass `current_config` from the caller?

Reply:
yeah, was just trying to minimize the signature changes... annoying that the dai that I want to actually access is only "dug out" in the ipc sending function. Actually I just want to set the config number related members to dai -> but need to pass down those with quite a route...

Reply:
ok